src/main/java/webserver/http/model/HttpRequest.java

```diff
@@ -0,0 +1,81 @@
+package webserver.http.model;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HttpRequest {
+
+    private final RequestLine requestLine;
+    private final RequestHeaders requestHeaders;
+    private RequestBody requestBody;
+
+    public HttpRequest(BufferedReader bufferedReader) throws IOException {
+        String line = bufferedReader.readLine();
+        this.requestLine = new RequestLine(line);
+        this.requestHeaders = new RequestHeaders(bufferedReader, line);
+        this.requestBody = null;
+        if (Method.isPost(requestLine.getMethod())) {
+            this.requestBody = new RequestBody(bufferedReader, requestHeaders);
+        }
+    }
+
+    public HttpRequest(String httpRequestText) {
+        List<String> textList = Arrays.asList(httpRequestText.split("\n"));
+        List<List<String>> requestInformation = requestHeaderAndBody(textList);
+
+        this.requestLine = new RequestLine(textList.get(0));
+        this.requestHeaders = new RequestHeaders(requestInformation.get(0));
+        this.requestBody = null;
```

Comment:
```suggestion
        this.requestLine = textList.getRequestLine();
        this.requestHeaders = requestInformation.getRequestHeaders();
        this.requestBody = RequestBody.empty();
```

이건 취향차이라서 꼭 적용해주실 필요는 없지만,
index 로 직접 접근하기 보다는 일급 컬렉션 객체를 따로 정의해서, 메서드명으로 가져오고 싶은게 무엇인지 명시하는 건 어떠신가요?

또한 NullException 을 예방하기 위해, null 을 직접 할당하기 보다는 별도의 empty 싱글톤 객체를 재활용하는 코드 스타일도 있습니다.
참고로 정말로 취향의 문제라서 꼭 적용해주시지 않아도 괜찮습니다.

Reply:
인덱스를 바로 사용하는 것에 약간 저도 헷갈리던 부분이 있었는데
일급 컬렉션을 사용하면 더 좋은 것 같습니다.
해당 내용 3단계에서 수정하여 반영 하였습니다!